server.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+// Server is an interface implemented by a value that can provide values to a
+// handler in a separate goroutine.
+type Server interface {
+	// Close prevents the server from sending any more data to a handler.
+	// A call to this method will block until the server has finished
+	// processing all existing data.
+	// This method will fail if the server is already closed.
+	Close() error
+}
+
+// A UnixServer represents a listener on a unix socket that forwards data to a
+// handler.
+// This server expects data sent over a connection to be terminated with a '\n'
+// symbol.
+// The result from the handler will be sent back across the TCP connection,
+// with each entry (assignment and location) terminated with a '\n' symbol.
+type UnixServer struct {
+	listener  *net.UnixListener
+	waitGroup *sync.WaitGroup
+	schan     chan time.Time
+}
+
+// TODO: Comply with Close()
+
+// NewUnixServer creates a UnixServer that listens on the given file, passing
+// incoming traffic to the given handler.
+func NewUnixServer(monitor Monitor, handler Handler, filename string) (*UnixServer, error) {
```

Comment:
Maybe just the naming, but this API seems a bit unintuitive to me. So I'll do:

srv, err = NewUnixServer(mon, h, fn)

and it starts serving right away? I think it might be clearer if I do `srv.Start()` or something similar after checking that `err` is nil